test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1178,7 +1178,11 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
           .to.be.instanceOf(Error)
           .to.have.property('name', 'MongoServerSelectionError');
 
-        expect(insertError).to.match(/connect ECONNREFUSED 127.0.0.1:27021/);
+        expect(insertError, 'Error must contain ECONNREFUSED').to.satisfy(
+          error =>
+            /ECONNREFUSED/.test(error.message) ||
+            error.cause.cause.errors.every(e => e.code === 'ECONNREFUSED')
```

Comment:
will this fail gracefully if the message doesn't match but also there is no `cause.cause.errors` property? I'm not sure what happens if you throw inside a `satisfy` clause like that

Reply:
I tested locally and the answer is "no", so as a follow up, can we make sure that this does fail gracefully with a reasonable message?

Reply:
optional chaining to the rescue, fixed